dora/stdlib/Iterator.dora

```diff
@@ -0,0 +1,4 @@
+trait Iterator {
+  fun hasNext() -> Bool;
+  fun next() -> Self;
```

Comment:
I think with this return type the impls for `IntRangeIter` and `CodepointIterator` should not compile, but they do.

I guess we are missing a check there, @dinfuehr?

Reply:
This is indeed surprising, most likely a bug.

Reply:
Shall I remove the commit from this PR for now?

Reply:
Yes.

Reply:
Ah, maybe better: I kept the file move and just undid the CodepointIterator impl and the result type change. Also acceptable?

Reply:
Repro for bug is https://github.com/dinfuehr/dora/issues/164